Step5 - 로또(수동) PR 요청드립니다.

KangJunJun opened the pull request.

KangJunJun committed:
docs: README.md Step5 기능목록 추가

KangJunJun committed:
feat: 수동으로 구매할 개수 입력 기능

- 관련 클래스의 단위 테스트 리펙터링
- 음수인경우 0으로 바꾸던 걸 예외처리로 변경

KangJunJun committed:
refactor: getter 제거 및 코드스타일 수정

- 우테코 코드템플릿 적용
- getter 제거하며 객체 의존주입 변경

KangJunJun committed:
feat: 수동으로 번호를 입력할 수 있는 기능구현

KangJunJun committed:
refactor: WinningNumber 클래스를 Lotto에 합병

- 코드템플릿 추가 적용

KangJunJun committed:
feat: 수동로또와 자동로또가 합쳐서 출력되도록 수정

KangJunJun committed:
fix: 구매액을 초과하는 수동 개수 입력에 대한 예외처리 수정

joojimin commented on README.md:
```diff
@@ -44,12 +44,29 @@
+* 수동으로 구매할 개수 입력 기능
+* 수동으로 번호를 입력할 수 있는 기능
+* 자동 생성 개수에서 수동을 개수를 제외하도록 수정
+* 수동로또와 자동로또가 합쳐서 출력되도록 수정
```
요구사항 정리 👍

joojimin commented on src/main/java/step3/ValidationUtils.java:
```diff
@@ -4,21 +4,35 @@ public class ValidationUtils {
     }
-    
+
     private static void validLottoNumber(int parseNumber) {
```
ValidationUtils라는 이름에서 일반적인 Util성 ( parseInt 메서드 )정도만 모여있는게 좋지 않을까 생각이들어요!
도메인에 한정되는 validate 로직까지 전부 모여있다면 너무많은 관심사가 한 클래스에 모일거같습니다! 
준호님 생각은 어떠실까요~?

joojimin commented on src/main/java/step3/domain/LotteryTicket.java:
```diff
@@ -5,22 +5,19 @@
-    public LotteryTicket(int payment) {
-        this.payment = new Payment(payment);
+
+    public LotteryTicket() {
```
public한 기본생성자는 정의안해도 좋을거같아요!

joojimin commented on src/main/java/step3/domain/Lotto.java:
```diff
@@ -1,51 +1,82 @@
     private int matchCount;
-    
+
     public Lotto(List<Integer> lottoNumbers) {
```
`규칙 3: 모든 원시값과 문자열을 포장한다.` 에 따라서 Lotto안에 들어가는 각각의 LottoNumber들도 래핑해보면 어떨까요~?
또 추가로 래핑하다보면 해당 클래스에서 캡슐화할 수 있는 로직들도 보일 것 같아요!

joojimin commented on src/main/java/step3/domain/Lotto.java:
```diff
@@ -1,51 +1,82 @@
         return winningBonusNumber.checkBonusNumber(lottoNumbers);
     }
+
+    public List<Integer> getWinningNumbers() {
```
WinningNumbers를 감쌀수 있는 일급컬렉션을 만들어보면 어떨까요!
앞서 리뷰드린 LottoNumber들과 마찬가지로 캡슐화할 수 있는 로직도 생길 것 같습니다!

joojimin reviewed (changes requested):
미션진행 고생하셨습니다 준호님~!
마지막 미션이다보니 조금 빡빡하게 코멘트 드렸어요!
확인 한번 부탁드립니다~!

KangJunJun committed:
refactor: public 한 기본생성자 제거

KangJunJun committed:
refactor: LottoNumber 클래스 추가하여 래핑

- Lotto 안에 로또번허들을 LottoNumber로 래핑
- WinningNumbers 역시 LottoNumber로 래핑
- valid체크는 LottoNumber 안에서 처리

https://github.com/next-step/java-lotto-pro/pull/888#discussion_r1012750922

KangJunJun committed:
refactor: 불필요한 getter 제거

KangJunJun committed:
refactor: 사용하지 않는 메서드 제거

KangJunJun committed:
refactor: ValidationUtils를 Utils로 변경

- Utils 에는 도메인과 관련 없는 함수로만 정리
- 각 도메인에 필요한 valid 함수는 도메인으로 이동

joojimin reviewed (approved):
피드백 반영 고생하셨습니다~!!
코드가 한결 깔끔해진것 같아요!!
앞으로 남은 미션도 화이팅입니다!!
